content/3_Silver/Binary_Search_Ans.mdx

```diff
@@ -328,6 +365,20 @@ public static int lstTrue(int lo, int hi) {
 
 </JavaSection>
 
+<PySection>
+
+```python
+def lstTrue(f, lo, hi): # python scoping...
+    lo-=1
+    locs=locals()
```

Comment:
I'm a little bit confused about where all the local and global things came from, and why this is necessary. If it is necessary, it needs to be explained more (because if I'm confused, then they'll probably be very confused). If it can be taken out, it would be a good idea to.

Reply:
Oh this is just some weird python scoping rules..
If we want to use python ternary expressions like (a = b if cond else c) here then we need exec and those locales.
You can just remove this python code since that python ternary expression only accept expressions but not statements.

Reply:
yea i see what you mean...
honestly this implementation is far too cumbersome to be worth it (ternary should shorten the code not lengthen it), so you have 2 options:
1. leave out the python implementation for this (or add a small blurb stating reasons why there isn't a python imp)
2. put a small blurb above the code basically recommending people to not do this in python, why people shouldn't do this, and having the code there for anyone who is curious.
either option is fine but as it is right now it's counterintuitive

Reply:
sure